src/SqlPersistence/SynchronizedStorage/SqlDialect_MsSqlServer.cs

```diff
@@ -49,13 +48,12 @@ public partial class MsSqlServer
                 if (ambientTransaction == null)
                 {
                     // Other modes handled by creating a new session.
-                    return null;
+                    return (WasAdapted: false, Connection: null, Transaction: null, OwnsTransaction: false);
                 }
 
                 var connection = await connectionManager.OpenConnection(context.GetIncomingMessage(), cancellationToken).ConfigureAwait(false);
                 connection.EnlistTransaction(ambientTransaction);
-
-                return storageSessionFactory(connection, null, true);
+                return (WasAdapted: true, Connection: connection, Transaction: null, OwnsTransaction: false);
```

Comment:
```suggestion
                return (WasAdapted: true, Connection: connection, Transaction: null, OwnsTransaction: true);
```

Because we create a connection here and the `StorageSession` needs to Dispose it and it only calls `Dispose` if `OwnsTransaction` is `true`. Because the `Transaction` part of the tuple is `null`, it will not commit the transaction and that is fine because in this case the (ambient) transaction is managed by the transport.